Provide types for typescript

glorat opened the pull request:
* Create index.d.ts

* Create vue.d.ts

* Update package.json

glorat commented:
Relates to issue https://github.com/pulsardev/vue-tour/issues/73

I've tested this locally against my consuming project but I only use very little of the full API so I've correspondingly kept the exposed interface small and to the methods I'm confident have been declared correctly.

Snapu commented on types/index.d.ts:
```diff
@@ -0,0 +1,14 @@
+  stop(): void
+  skip(): void
+  finish(): void
+  currentStep: number
```
Would be nice to also include the computed property `isRunning`. E.g. I use it to ensure my tours do not overlap.

glorat replied:
I've added that to the PR now. Please test by putting 
`"vue-tour": "git://github.com/glorat/vue-tour.git#typings",`

in the dependencies section of your package.json and let me know if that works for you

HZooly commented:
Thanks for this @glorat, we will check the PR with the team as soon as possible!

glorat commented:
@Snapu and anyone else looking to test (and give feedback!) before final review and merge, you can add this to your dependencies in package.json to test the typings
`"vue-tour": "git://github.com/glorat/vue-tour.git#typings"`

Otherwise, "it works for me" :)

Snapu commented:
> @Snapu and anyone else looking to test (and give feedback!) before final review and merge, you can add this to your dependencies in package.json to test the typings
> `"vue-tour": "git://github.com/glorat/vue-tour.git#typings"`
> 
> Otherwise, "it works for me" :)

Just tested, it works for me too 👍

Snapu commented on types/index.d.ts:
```diff
@@ -4,11 +4,18 @@ import Vue from 'vue';
   currentStep: number
+
+  // Computed
+  isRunning(): boolean
```
This is not working. You have to remove `()` for the computed props.
E.g.:
```
isRunning: boolean
```

glorat replied:
Thanks for the feedback. I've pushed the update. Let me know if that works better.

Snapu replied:
Now it works. I have tested 👍

Outpox commented:
Thanks for the PR, your investment is greatly appreciated!  
I only have one struggle with it, you haven't followed the commit naming guidelines: https://github.com/pulsardev/vue-tour/blob/master/CONTRIBUTING.md#git-commit-guidelines  
This allows us to easily generate changelogs and have a streamlined history.  
@mmorainville won't be pleased 🤣  

My suggestion here would be: follow the guidelines from now on and we'll rename theses commits (without changing the commits authorship obviously).

glorat commented:
> Thanks for the PR, your investment is greatly appreciated!
> I only have one struggle with it, you haven't followed the commit naming guidelines: https://github.com/pulsardev/vue-tour/blob/master/CONTRIBUTING.md#git-commit-guidelines
> This allows us to easily generate changelogs and have a streamlined history.
> @mmorainville won't be pleased 🤣
> 
> My suggestion here would be: follow the guidelines from now on and we'll rename theses commits (without changing the commits authorship obviously).

I learned something new today! I should consider adopting the same guidelines in some of my projects... release notes are a pain

I've squashed and renamed my PR

Outpox commented:
Thanks for your reactivity and sorry to nitpick (😅) but the [subject chapter](https://github.com/pulsardev/vue-tour/blob/master/CONTRIBUTING.md#subject) states: `don't capitalize first letter`

glorat commented:
> Thanks for your reactivity and sorry to nitpick (😅) but the [subject chapter](https://github.com/pulsardev/vue-tour/blob/master/CONTRIBUTING.md#subject) states: `don't capitalize first letter`

Fixed (and fixed in my other repo where I start applying this standard). Thanks for the education

speasley commented:
Hello, I'm wondering what the status of this PR is.

speasley commented:
Is this project no longer maintained?

HZooly commented:
Hello @speasley, please remember that we're doing this project during our free time.
Thanks to @glorat, types seem to be in a good progress, but we have to review it, test it etc.

We have a meeting tomorrow, we will talk about it 🚀

HZooly commented:
@glorat @Snapu @speasley [Vue Tour 1.3.1](https://www.npmjs.com/package/vue-tour/v/1.3.1) is published with Types 🥳
